### PR TITLE
Add quantization c api

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

**Comment by scxiao** [comment]
@pfultz2 @kahmed10 When you guys have time, can you take a look at this PR? Thanks!

**Comment by kahmed10** [comment]
It looked fine to me, aside from what we discussed with possibly changing `const char* const* names` that @pfultz2 mentioned

File: src/api/include/migraphx/migraphx.h [hunk 1]
```diff
@@ -70,6 +70,12 @@ typedef const struct migraphx_program* const_migraphx_program_t;
 typedef struct migraphx_onnx_options* migraphx_onnx_options_t;
 typedef const struct migraphx_onnx_options* const_migraphx_onnx_options_t;
 
+typedef struct migraphx_op_names* migraphx_op_names_t;
+typedef const struct migraphx_op_names* const_migraphx_op_names_t;
```
Comment (pfultz2) [inline src/api/include/migraphx/migraphx.h:74]: Can we name this `migraphx_quantize_op_names`?

File: src/api/include/migraphx/migraphx.h [hunk 2]
```diff
@@ -197,6 +203,43 @@ migraphx_status migraphx_parse_onnx_buffer(migraphx_program_t* out,
...
 
+migraphx_status migraphx_op_names_destroy(migraphx_op_names_t op_names);
+
+migraphx_status migraphx_op_names_create(migraphx_op_names_t* op_names);
+
+migraphx_status migraphx_op_names_add(migraphx_op_names_t op_names, const char* name);
+
+migraphx_status migraphx_op_names_size(size_t* out, migraphx_op_names_t op_names);
+
+migraphx_status migraphx_op_names_get(const char** out, migraphx_op_names_t op_names, size_t idx);
```
Comment (pfultz2) [inline src/api/include/migraphx/migraphx.h:214]: I dont think we need that `size` and `get` method.

File: src/api/include/migraphx/migraphx.h [hunk 3]
```diff
@@ -197,6 +203,43 @@ migraphx_status migraphx_parse_onnx_buffer(migraphx_program_t* out,
...
+
+migraphx_status migraphx_calibration_data_add_element(migraphx_calibration_data_t calibration_data,
+                                                      migraphx_program_parameters_t elem);
+
+migraphx_status migraphx_quantize_int8(migraphx_program_t prog,
+                                       migraphx_target_t target,
+                                       migraphx_calibration_data_t data,
+                                       migraphx_op_names_t name);
+
+migraphx_status migraphx_quantize_int8_default(migraphx_program_t prog,
```
Comment (pfultz2) [inline src/api/include/migraphx/migraphx.h:239]: Maybe we should add a struct to store `migraphx_calibration_data_t` and `migraphx_op_names_t`. This way when we add more options we dont need to add new functions.

File: src/api/include/migraphx/migraphx.h (hunk 1 above)
Comment (scxiao) [reply src/api/include/migraphx/migraphx.h:74]: renamed

File: src/api/include/migraphx/migraphx.h (hunk 2 above)
Comment (scxiao) [reply src/api/include/migraphx/migraphx.h:214]: removed

File: src/api/include/migraphx/migraphx.h (hunk 3 above)
Comment (scxiao) [reply src/api/include/migraphx/migraphx.h]: added an structure quantize_options to include op_names and calibration_data.

File: src/api/include/migraphx/migraphx.h [hunk 4]
```diff
@@ -197,6 +203,32 @@ migraphx_status migraphx_parse_onnx_buffer(migraphx_program_t* out,
...
+migraphx_status migraphx_quantize_op_names_destroy(migraphx_quantize_op_names_t quantize_op_names);
+
+migraphx_status migraphx_quantize_op_names_create(migraphx_quantize_op_names_t* quantize_op_names);
+
+migraphx_status migraphx_quantize_op_names_add(migraphx_quantize_op_names_t quantize_op_names,
+                                               const char* name);
+
+migraphx_status migraphx_quantize_fp16(migraphx_program_t prog, migraphx_quantize_op_names_t name);
+
+migraphx_status migraphx_quantize_fp16_default(migraphx_program_t prog);
```
Comment (pfultz2) [inline src/api/include/migraphx/migraphx.h:215]: Can we shorten this to `migraphx_quantize_fp16` and call the other overload `migraphx_quantize_fp16_with_names`.

File: src/api/include/migraphx/migraphx.h [hunk 5]
```diff
@@ -197,6 +203,32 @@ migraphx_status migraphx_parse_onnx_buffer(migraphx_program_t* out,
...
+
+migraphx_status migraphx_quantize_options_destroy(migraphx_quantize_options_t quantize_options);
```
Comment (pfultz2) [inline src/api/include/migraphx/migraphx.h:217]: Maybe this should be named `migraphx_quantize_int8_options_t` since its only for int8 quantization.

Comment (scxiao) [reply src/api/include/migraphx/migraphx.h:217]: renamed

File: src/api/include/migraphx/migraphx.h (hunk 4 above)
Comment (scxiao) [reply src/api/include/migraphx/migraphx.h]: fixed

**Comment by scxiao** [comment]
Looks like all the missing code are the throw exceptions, so it is not easy to create unit tests to test these lines.
